pkg/kgo/compression.go

```diff
@@ -266,15 +268,33 @@ type zstdDecoder struct {
 	inner *zstd.Decoder
 }
 
-func (d *decompressor) decompress(src []byte, codec byte) ([]byte, error) {
+func (d *decompressor) decompress(src []byte, codec byte, pool *pool.BucketedPool[byte]) ([]byte, error) {
 	// Early return in case there is no compression
 	compCodec := codecType(codec)
 	if compCodec == codecNone {
 		return src, nil
 	}
-	out := byteBuffers.Get().(*bytes.Buffer)
-	out.Reset()
-	defer byteBuffers.Put(out)
+	var (
+		out *bytes.Buffer
+		decodedBufSize int
+	)
+
+	if compCodec == codecSnappy {
+		var err error
+		decodedBufSize, err = s2.DecodedLen(src)
+		if err != nil {
+			return nil, err
+		}
+	} else {
+		// Make a guess at the output size.
+		decodedBufSize = len(src) * 2
+	}
+
+	outBuf := pool.Get(decodedBufSize)[:0]
+	defer func() {
+		pool.Put(outBuf)
+	}()
```

Comment:
```suggestion
	defer pool.Put(outBuf)
```

doesn't this work?

Reply:
I must have accidentally overwritten the previous change. 😮‍💨

Reply:
_addressed @ 88058a005e32882ad66071dbf73f15eb71374cbd_